docs/content/doc/installation/database-preparation.en-us.md

```diff
@@ -0,0 +1,153 @@
+---
+date: "2019-01-16"
+title: "Database Preparation"
+slug: "database-prep"
+weight: 10
+toc: true
+draft: false
+menu:
+  sidebar:
+    parent: "installation"
+    name: "Database preparation"
+    weight: 20
+    identifier: "database-prep"
+---
+
+You need a database to use Gitea. Gitea supports PostgreSQL, MySQL, SQLite, and MSSQL. This page will guide into preparing database. Only PostgreSQL and MySQL will be covered here since those database systems are widely-used in production.
+
+Database instance can be on same machine as Gitea (local database setup), or on different machine (remote database).
+
+## MySQL
+
+1.  On database instance, login to database console as root:
+
+    ```
+    mysql -u root -p
+    ```
+
+    Enter the password as prompted.
+
+2.  Create database user which will be used by Gitea, authenticated by password.
+
+    For local database:
+
+    ```sql
+    SET old_passwords=0;
+    CREATE USER 'gitea' IDENTIFIED BY 'gitea';
+    ```
+
+    For remote database:
+
+    ```sql
+    SET old_passwords=0;
+    CREATE USER 'gitea'@'12.34.56.78' IDENTIFIED BY 'gitea';
+    ```
+
+    where `12.34.56.78` is the IP address of your Gitea instance.
+
+    Replace username and password above as appropriate.
+
+3.  Create database with UTF-8 charset and collation. If you expected to use emojis, use `utf8mb4` charset instead of `utf8`. Also, collation chosen depending on your expected content. When in doubt, use either `unicode_ci` or `general_ci`.
+
+    ```sql
+    CREATE DATABASE 'giteadb' CHARACTER SET 'utf8' COLLATE 'utf8_unicode_ci';
```

Comment:
`utf8` is bad advice. I would not even mention it, just make it `utf8mb4`.

Reply:
@silverwind why bad, although gitea supports it?

Reply:
because utf8 isn't actually utf8 and is missing a significant number of characters.

Reply:
also most of the issues users have with mysql & gitea are due to them using just `utf8` instead of `utf8mb4`